react/components/EditorContainer/StoreEditor/Styles/StyleEditor/typography/TypeToken.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { Dispatch } from 'react'
+import { RouteComponentProps } from 'react-router'
+import { Dropdown, Spinner } from 'vtex.styleguide'
+
+import { FormattedMessage } from 'react-intl'
+import ListFontsQuery, {
+  FontFamily,
+  ListFontsQueryResult,
+} from '../queries/ListFontsQuery'
+import StyleEditorHeader from '../StyleEditorHeader'
+import { prettify, prettifyCamelCase } from '../utils/typography'
+
+interface WrapperProps extends RouteComponentProps<TypeTokenParams> {
+  config: TachyonsConfig
+  onSave: () => void
+  updateConfig: (config: DeepPartial<TachyonsConfig>) => void
+}
+
+interface Props extends WrapperProps {
+  result: ListFontsQueryResult
+}
+
+const TypeTokenWrapper: React.FunctionComponent<WrapperProps> = props => {
+  return (
+    <ListFontsQuery>
+      {result => <TypeToken {...props} result={result} />}
+    </ListFontsQuery>
+  )
+}
+
+const TypeToken: React.FunctionComponent<Props> = ({
+  match,
+  result: { loading, data, error },
+  config,
+  onSave: saveStyle,
+  updateConfig,
+}) => {
+  const token = match.params.id as keyof TypographyStyles
+  const { [token]: font } = config.typography.styles
+
+  if (loading) {
+    return (
+      <>
+        <StyleEditorHeader title={prettify(token)} />
+        <div className="pv8 flex justify-center">
+          <Spinner />
+        </div>
+      </>
+    )
+  }
+
+  if (error != null || data == null) {
+    // TODO: add error state
+    return <div />
+  }
+
+  const dispatch: Dispatch<Partial<Font>> = (fontChange: Partial<Font>) =>
+    updateConfig({ typography: { styles: { [token]: fontChange } } })
+
+  const saveLabel = (
+    <FormattedMessage
+      id="admin/pages.admin.pages.form.button.save"
+      defaultMessage="Save"
+    />
+  )
+
+  return (
+    <>
+      <StyleEditorHeader
+        title={prettify(token)}
+        buttonLabel={saveLabel}
+        onButtonClick={saveStyle}
+      />
+      <div className="ph6">
+        {Object.keys(font)
+          .filter(key => !key.startsWith('__'))
+          .map(key => (
+            <TypeTokenEntry
+              font={font}
+              dispatch={dispatch}
+              fontFamilies={data.listFonts}
+              id={key as keyof Font}
+            />
+          ))}
+      </div>
+    </>
+  )
+}
+
+function getOnChange(key: keyof Font, dispatch: Dispatch<Partial<Font>>) {
+  return (_: React.ChangeEvent, value: string) => dispatch({ [key]: value })
+}
+
+interface DropdownOption {
+  value: string
+  label: string
+}
+
+function getOptions(
+  key: keyof Font,
+  font: Font,
+  fontFamilies: FontFamily[]
+): DropdownOption[] {
+  const options = (() => {
+    switch (key) {
+      case 'fontFamily':
+        return fontFamilies.map(({ id, fontFamily }) => ({
+          label: fontFamily,
+          value: fontFamily,
+        }))
+      case 'fontWeight':
+        return [
+          { label: 'Thin', value: '100' },
```

Comment:
Maybe i18n these options :S

Reply:
I thought about it, but I left like this to be as close as the actual css values as possible.